### ISSUE TITLE
BLISS assigns low probability of detection to bright objects (mag <19) 

### ISSUE THREAD

**Issue by ismael-mendoza** [prob-ml/bliss#547]
As highlighted by the most recent detection metric figure we miss 25 out of 50 galaxies at < 19 magnitude. How big is the probability miscalibration issue? 

![Screen Shot 2022-04-18 at 9 28 24 PM](https://user-images.githubusercontent.com/11745764/163902020-6f4f8e94-5586-4db0-acb8-4deaa2d14444.png)

**Comment by jeff-regier** [comment]
I think Derek reported that a similar fraction of 19 mag galaxies are not detected in semi-synthetic data. That's good news because it means the issue is not due to model misspecification, which could have been tricky to fix.

My question: by the end of sleep-phase training of the location encoder, how often do we fail to detect a 19 magnitude galaxy? It can't be 50% of the time, right? And if not, why would we miss a large proportion of the 19 magnitude galaxies in semi-synthetic data?

**Comment by dereklhansen** [comment]
Here are results with a 10% detection ratio (moving the threshold back to where it was before we adjusting the probabilities):

```
output-with-photo/sdss_figures
bliss
detections
           recall  expected_recall  precision  expected_precision    n  n_galaxies  n_galaxies_detected
18       0.968750              NaN   0.966667                 NaN   32          10                    9
19       0.875000              NaN   0.900000                 NaN   56          29                   22
20       0.847059              NaN   0.836066                 NaN   85          59                   48
21       0.801136              NaN   0.814815                 NaN  176         132                   97
22       0.711297              NaN   0.613419                 NaN  239         158                   90
23       0.504425              NaN   0.206439                 NaN  113          47                    4
overall  0.792867         0.669812   0.511936            0.492125  729         442                  308
```

We detect many more of the bright galaxies (31/39) in the image.

Now, if we apply the hill-climbing algorithm in #552  which utilizes the true posterior (i.e. the generative model prior and likelihood) targeting a 50% detection ratio, here's what we get:

```
output-hc/sdss_figures
bliss
detections
           recall  expected_recall  precision  expected_precision    n  n_galaxies  n_galaxies_detected
18       0.968750              NaN   0.966667                 NaN   32          10                    9
19       0.875000              NaN   0.900000                 NaN   56          29                   22
20       0.847059              NaN   0.836066                 NaN   85          59                   48
21       0.789773              NaN   0.813433                 NaN  176         132                   95
22       0.577406              NaN   0.728745                 NaN  239         158                   58
23       0.371681              NaN   0.355000                 NaN  113          47                    0
overall  0.724280              NaN   0.717391                 NaN  729         442                  269
```

Not only do we get a better overall result, notice that none of the brightest objects get removed during the algorithm, indicating that the true MAP favors including them.

**Comment by dereklhansen** [comment]
For reference, here are the BLISS estimates on semi-synthetic figures (50% detection threshold):
```
output-with-photo/semisynthetic_figures
bliss
detections
           recall  expected_recall  precision  expected_precision    n  n_galaxies  n_galaxies_detected
18       0.971429              NaN   1.000000                 NaN   35          13                   13
19       0.859375              NaN   1.000000                 NaN   64          37                   29
20       0.730769              NaN   1.000000                 NaN   78          52                   35
21       0.676692              NaN   0.975610                 NaN  133          89                   55
22       0.493274              NaN   0.942623                 NaN  223         142                   45
23       0.173410              NaN   0.704918                 NaN  173         107                   10
overall  0.555405         0.640559   0.942661            0.885355  740         453                  200
```

For bright galaxies, these look more like the numbers after using the HC algorithm on real data. So my impression is there is model misspecification in the real data, and my suspected culprit is the background (I made some slides for the meeting today).

**Comment by jeff-regier** [comment]
Interesting results, I'm looking forward to hearing about the HC algorithm. Is there anything unusual about eight galaxies we don't detect according to the row below? For example, are they all blends? Or are they all close to the border between tiles? Or do they all have high ellipticity / large half-light radius?
```
19       0.859375              NaN   1.000000                 NaN   64          37                   29
```

**Comment by ismael-mendoza** [comment]
I added some examples of undetections (along with their probabilities) to my meetings slides for today. These are on real data though not semi-synthetic. We can also look through the plotly in case you have specific location in mind

**Comment by dereklhansen** [comment]
We see quite a bit higher recall for brighter galaxies on synthetic data sampled with the training rate (`0.03`).

This suggests to me that we might want to consider sleep phase training with the true rate seen on the real data (`0.004`). It might not be enough to do a post-hoc adjustment.

```
output-with-photo/simulated_figures
bliss
detections
           recall  expected_recall  precision  expected_precision     n  n_galaxies  n_galaxies_detected
18       0.976303              NaN   0.989950                 NaN   211         144                  139
19       0.947214              NaN   0.964968                 NaN   341         235                  219
20       0.879875              NaN   0.965318                 NaN   641         436                  369
21       0.805221              NaN   0.925444                 NaN   996         702                  522
22       0.627344              NaN   0.861369                 NaN  1653        1183                  623
23       0.277253              NaN   0.698830                 NaN  1165         862                  122
overall  0.677667         0.622148   0.894857             0.88452  5212        3682                 2175
```

**Comment by dereklhansen** [comment]
Closed by #561, but @ismael-mendoza you'll need to retrain the LocationEncoder in `case_studies/sdss_galaxies`
